app/services/base_auth_service.rb

```diff
@@ -6,13 +6,42 @@ class BaseAuthService
   include HTTParty
 
   class << self
-    def find_user_by_token(access_token)
+    def user_existence_validation(create_user, users)
+      raise ::UserAlreadyExist::Error if user_already_exist(create_user, users)
+      raise ::UserNotFound::Error if user_not_found(create_user, users)
+    end
+
+    def first_or_initialize_user(primary_key, id, user_atts)
+      user = User.where(primary_key => id).first_or_initialize
+      user.update!(user_atts)
```

Comment:
`first_or_initialize_user` is not really an accurate name because of this `update!` call. `first_or_create_user` would be more accurate.